### DISCUSSION TITLE
telegram.error.BadRequest: Can't invite members to a private chat

### DISCUSSION THREAD

**Discussion by p17m0** [python-telegram-bot/python-telegram-bot#3493] [Q&A] [answered]
Hello everyone!
I have a problem with my private group.
I added my bot to a group and gave him all rights.
Also i copied ChatMemberBotExample and create Handler for making chat invite link and it doesn't work
Problem: telegram.error.BadRequest: Can't invite members to a private chat
![image](https://user-images.githubusercontent.com/81479851/211193748-c0282eb4-35d7-481c-a360-523aedff8cbf.png)
![image](https://user-images.githubusercontent.com/81479851/211193795-bd8984ec-3423-4452-8132-e22087e9c9d5.png)

**Comment by Bibo-Joshi** [comment, marked as answer]
Double check that `id_group` is the correct chat id of a group chat or channel. it must be a negative integer.

**Reply by p17m0** [reply]
I see, but: telegram.error.BadRequest: Chat not found

**Reply by Bibo-Joshi** [reply]
then either the id is still wrong or the bot does not have access to that chat

**Reply by p17m0** [reply]
id is correct
![image](https://user-images.githubusercontent.com/81479851/211195732-6f130814-741d-48a1-9310-b24b701ba48d.png)
and bot have all acces in the group as admin

**Reply by p17m0** [reply]
However /show_chats doesn't show this chat if i reload my bot

**Reply by Bibo-Joshi** [reply]
> id is correct ![image](https://user-images.githubusercontent.com/81479851/211195732-6f130814-741d-48a1-9310-b24b701ba48d.png) and bot have all acces in the group as admin

please don't use clients to get the chat id, but rather read it from the `Updates` that your bot gets. If your group is a supergroup, the id will start with a -100. clients tend to don't show that -100.

> However /show_chats doesn't show this chat if i reload my bot

Because the `chatmemberbot` example doesn't use persistence, i.e. the in-memory data is lost upon restart

**Reply by p17m0** [reply]
Thank you!

**Reply by jashjasani** [reply]
@Bibo-Joshi Is there a way i can add a user to telegram chat directly ?

**Reply by Bibo-Joshi** [reply]
Bots can't add new members to chats, no.
